src/dmd/backend/dwarfdbginf.d

```diff
@@ -1277,40 +1309,92 @@ static if (1)
         }
     }
 
+    /*************************************
+     * Add a directory to `lineDirectories`
+     */
+    uint dwarf_line_add_directory(const(char)* path)
+    {
+        assert(path);
+        return dwarf_line_add_directory(path[0 .. strlen(path)]);
+    }
+
+    /// Ditto
+    extern(D) uint dwarf_line_add_directory(const(char)[] path)
```

Comment:
This function accepts a slice, but its implementation requires that path be 0 terminated.